backend/src/utils/proxy.ts

```diff
@@ -37,7 +37,7 @@ export const proxyService =
         internalPort: number | string;
         prefix?: string;
         suffix?: string;
-        namespace?: string;
+        namespace?: string | ((fastify: KubeFastifyInstance) => Promise<string>);
```

Comment:
It looks like you never await the function version of this in the usage below. This means when we use `namespace` in string templates below we actually get the string `"[Function: getModelRegistryNamespace]"` instead of the namespace. We need to check the type and await the function if it is not a string - or if we want to avoid run-time type checking we could just change this to `getNamespace` and always take it as a function to be awaited, and then in the usages that currently pass a string they can pass `() => Promse.resolve("some-namespace")`.

This is a bit more complex actually... `fastify.register()` needs the namespace as part of its `prefix` param, so we need to check if `service.namespace` is a function and await it before we even call `fastify.register()`. I assume that will still work since the function is async, but I'm not actually certain. It'll need testing.